Automattic/wp-calypso
# Post-purchase: Add a modal asking for confirmation when the user wants to cancel a migration

Closes to #96237 

## Proposed Changes
* Add a modal asking for confirmation when the user wants to cancel a migration. 
* Adjust the `ConfirmModal` component to follow the new Design System.

https://github.com/user-attachments/assets/6f9b64d9-9993-492e-9514-940d1dda9f71



## Why are these changes being made?
* We want to prevent the user from canceling pending migration unintentionally

## Testing Instructions
* Create a pending migration following the instructions `How to create a pending migration.` 
* Go to /sites
* Select the created site
* Click on the Cancel Migration CTA
* Check the confirm and the cancel button

> [!NOTE]  
> Please ignore the label `Pending migration` after the cancellation, it is a bug that will be  fixed by #96244

## How to create a pending migration
* Go to any migration flow (e.g./setup/migration)
* Enable the feature flag sessionStorage.setItem('flags', 'automated-migration/pending-status') and reload the page
* Follow all migration steps until you see the credential or setup instructions steps.
* Close the tab

## Pre-merge Checklist
